### PR TITLE
Add responsivity

### PR DESCRIPTION
**Pull request by vaclavpavlicek**
I created directive which is applied to some element (I think best is body). This directive broadcast window resize event and d3-cloud directive listens to this broadcasts. So when window resizes updateCloud method is called and cloud updates. What do you think about it? Is it OK or should I do something better? #22

### CODE REVIEW

**Comment by vaclavpavlicek** [comment]
I removed directive, because it's overkill. What do you think about it now?

**Comment by grtjn** [comment]
Thanks for taking the effort, looks pretty good actually. Can you just add a $scope on $destroy event to disable the resize listener? Something like:

```
scope.$on('$destroy', function {
  angular.element($window).off('resize', onResize);
});
```

**Comment by vaclavpavlicek** [comment]
Is it OK now?

**Comment by grtjn** [comment]
Looks great, thnx!

I'd like to run a quick test before I merge. I'm a little swamped with other work though..

**Comment by vaclavpavlicek** [comment]
Ok doesn't matter.

**Comment by vaclavpavlicek** [comment]
How is it going?

**Comment by grtjn** [comment]
Great timing, I was looking at it just now.. :)

**Comment by grtjn** [comment]
I noticed my gh-pages were running with angular 1.3, which does not support optional bindings as it turns out. Because of that the default filter didn't kick in, and I ended up with no items to show. The angular dependency should be raised to ^1.4.4 in bower.json.

I also noticed that the svg element on gh-pages was getting height="0". I added setting width and height attribs on that element in the update function of the directive to fix that. With that I think the svg width 100% would be redundant.

I'm also thinking that the update and draw function of the directive should be merged together, just not entirely sure how..

Last thing I noticed was that you were using binary operators in `if (!updateflag & items[i].name !== oldValue[i].name & items[i].score !== oldValue[i].score) {`. Overlooked that last time. Could you replace that with: `if (!updateflag && (items[i].name !== oldValue[i].name) && (items[i].score !== oldValue[i].score)) {`?

**Comment by vaclavpavlicek** [comment]
With svg width set to 100% X scrollbar won't appear. But without that style X scrollbar will appear but wordcloud will update. So should I really remove setting svg width to 100%?

**Comment by vaclavpavlicek** [comment]
I think having update and draw method split is better than one long method.

**Comment by vaclavpavlicek** [comment]
Is it better now?

**Comment by vaclavpavlicek** [comment]
How is it going?

**Comment by grtjn** [comment]
Holiday season. I'll get back to you today or tomorrow..

File: dist/d3-cloud-ng.css [hunk 1]
```diff
@@ -1,3 +1,6 @@
+.d3-cloud svg {
+  width: 100%;
+}
```
Comment (grtjn) [inline dist/d3-cloud-ng.css]: Not sure this is really needed, but perhaps not that harmful..

File: less/d3-cloud-ng.less [hunk 2]
```diff
@@ -1,4 +1,7 @@
 .d3-cloud {
+  svg {
+    width: 100%;
+  }
```
Comment (grtjn) [inline less/d3-cloud-ng.less]: Not sure this is really needed, but perhaps not that harmful..

**Comment by grtjn** [comment]
Looking better! Maybe the svg width isn't an issue, but to be sure, did you look into this as well?

> I also noticed that the svg element on gh-pages was getting height="0". I added setting width and height attribs on that element in the update function of the directive to fix that.

I think I just copied over these two lines into the update function, perhaps that solves scrollbar issues as well? Or maybe we should specify overflow with either visible or hidden?

https://github.com/grtjn/d3-cloud-ng/blob/master/src/d3-cloud.directive.js#L203

**Comment by vaclavpavlicek** [comment]
I see!

**Comment by vaclavpavlicek** [comment]
Do you is it better?

**Comment by grtjn** [comment]
Sorry for taking so long to come back to this, but was rather occupied with other work.

I decided to merge your work manually, and clean up the code base at the same time. I appreciate your work, it was very helpful.

I'm still a bit unsure about the need for the width 100% stuff, so I have left that out. Instead, the resize listener will change the svg size based on the container. If that still needs attention, could you create a jsfiddle that shows accurate what goes wrong?

Closing this PR..

**Comment by grtjn** [comment]
FYI, I just published release 0.9.9..

**Pull request closed without merging**